src/area.c

```diff
@@ -88,12 +93,20 @@ LIBUI_FUNCTION(create) {
 	}
 	area->event_key = keyEvent;
 
-	uiAreaHandler handler = {event_draw_cb, event_mouse_cb, event_mouseCrossed_cb,
-							 event_dragBroken_cb, event_key_cb};
-
-	uiArea *ui_area = uiNewArea(&handler);
+	// we need the handler to live in the heap, or it
+	// will cause undefined behaviour being deallocated
+	// from the stack whne this function terminate
+	uiAreaHandler *handler = malloc(sizeof(uiAreaHandler));
+	handler->Draw = event_draw_cb;
+	handler->MouseEvent = event_mouse_cb;
+	handler->MouseCrossed = event_mouseCrossed_cb;
+	handler->DragBroken = event_dragBroken_cb;
+	handler->KeyEvent = event_key_cb;
+
+	uiArea *ui_area = uiNewArea(handler);
 	area->area = ui_area;
 
+	/*
```

Comment:
Since we must keep the handler reference alive this code is the correct way to go, but we should save the reference somewhere in order to release the reference later, when the control is destroyed.

Reply:
Wasn't it stored into the handle? `&(area->area_ref)`

Reply:
I see two way: make a map from the `struct control_handler` to the `uiArea` handle_external, or otherwise use a custom structure instead of `struct control_handler`. 

This new struct layout must begin with a  `struct control_handler` field, so that any other function that expect a  `struct control_handler` will still work using this new struct.

After this first field, we can dd others to store the `struct area_handle` data and the napi_ref of it.

Reply:
> Wasn't it getting stored into the handle? &(area->area_ref)

Yes, I meant we should to keep a reference to the `struct area_handle`

Reply:
Oh, wait, we can just install the events as normal event of the UiArea.
The current control functions will then keep care to free them on destroy.
Give me some minutes...

Reply:
This works for the events:

```c
	struct event_t *mouseEvent = create_event(env, mouse, "onMouse");
	if (mouseEvent == NULL) {
		return NULL;
	}
	install_event(handle->events, mouseEvent);
```

```c
static void event_mouse_cb(uiAreaHandler *h, uiArea *a, uiAreaMouseEvent *e) {
	struct control_handle *handle;
	ctrl_map_get(&controls_map, a, &handle);
	fire_event(handle->events->head->next->event);
}
```

Reply:
> This works for the events:

I'm trying exactly this path 😃

Reply:
> Is struct area_handle needed then?

We are developing a sort of telepathy, I wrote the response before your question!

Reply:
> We are developing a sort of telepathy

😆 😆

Reply:
> We can remove struct area_handle.

do you do it?

Reply:
> do you do it?

Yes, I did.